ast/visit.go

```diff
@@ -96,6 +96,13 @@ func walk(v Visitor, x interface{}) {
 			}
 		case *Term:
 			Walk(w, ts)
+		case *Every:
+			if ts.Key != nil {
+				Walk(w, ts.Key)
+			}
+			Walk(w, ts.Value)
+			Walk(w, ts.Domain)
+			Walk(w, ts.Body)
```

Comment:
I think we should call `Walk(w, ts)` here so the visitor receives `ts` before the key/value/domain/body.